app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -84,7 +84,7 @@ EOF
   fi
 elif [ -f /etc/debian_version ]; then
   cat << EOF > /etc/apt/sources.list.d/foreman_registration.list
-<%= shell_escape @repo %>
+<%= shell_strip_eof @repo %>
 EOF
```

Comment:
How about using the `save_to_file` macro instead? That's exactly intended for this use case. So
```erb
<%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
```